app/src/test/java/org/mozilla/tv/firefox/architecture/KillswitchLinearLayoutTest.kt

```diff
@@ -0,0 +1,124 @@
+
+    @Before
+    fun setup() {
+        val activityController = Robolectric.buildActivity(Activity::class.java)
+        killswitchLayout = KillswitchLinearLayout(activityController.get())
+    }
+
+    @Test
+    fun `GIVEN requirements are not set WHEN visibility is set THEN visibility should always be gone`() {
+        // Don't set requirements
+        visibilityIsAlwaysGone()
+    }
+
+    @Test
+    fun `GIVEN all locales are allowed WHEN visibility is set THEN it should be applied`() {
+        killswitchLayout.setRequirements(true, KillswitchLocales.All)
+
+        visibilityIsApplied()
+    }
+
+    @Test
+    fun `GIVEN experiment is disabled WHEN visibility is set THEN visibility should always be gone`() {
+        killswitchLayout.setRequirements(false, KillswitchLocales.All)
+
+        visibilityIsAlwaysGone()
+    }
+
+    @Test
+    fun `GIVEN visibility is set WHEN requirements are set THEN visibility should match the earlier set values`() {
+        killswitchLayout.visibility = View.VISIBLE
+        assertEquals(View.GONE, killswitchLayout.visibility)
+
+        killswitchLayout.setRequirements(true, KillswitchLocales.All)
+        assertEquals(View.VISIBLE, killswitchLayout.visibility)
+
+        visibilityIsApplied()
+    }
+
+    @Test
+    fun `GIVEN locale and country are on the accepted list WHEN visibility is set THEN it should be applied`() {
+        val mockLocaleManager = mockk<LocaleManager>()
+        every { mockLocaleManager.getCurrentLocale(any()) } answers { Locale("es", "mx") }
+
+        killswitchLayout.localeManager = mockLocaleManager
+
+        killswitchLayout.setRequirements(true, KillswitchLocales.ActiveIn(
+            Locale("es", "mx"),
+            Locale("en", "us")
+        ))
+
+        visibilityIsApplied()
+    }
+
+    @Test
+    fun `GIVEN country is not on the accepted list WHEN visibility is set THEN it should be gone`() {
+        val mockLocaleManager = mockk<LocaleManager>()
+        every { mockLocaleManager.getCurrentLocale(any()) } answers { Locale("es", "pa") }
+
+        killswitchLayout.localeManager = mockLocaleManager
+
+        killswitchLayout.setRequirements(true, KillswitchLocales.ActiveIn(
+            Locale("es", "mx"),
+            Locale("en", "us")
+        ))
+
+        visibilityIsAlwaysGone()
+    }
+
+    @Test
+    fun `GIVEN no country is specified AND language is on the list WHEN visibility is set THEN it should be applied`() {
+        val mockLocaleManager = mockk<LocaleManager>()
+        every { mockLocaleManager.getCurrentLocale(any()) } answers { Locale("es", "pa") }
+
+        killswitchLayout.localeManager = mockLocaleManager
+
+        killswitchLayout.setRequirements(true, KillswitchLocales.ActiveIn(
+            Locale("es"),
+            Locale("en")
+        ))
+
+        visibilityIsApplied()
+    }
+
+    private fun visibilityIsAlwaysGone() {
+        killswitchLayout.visibility = View.VISIBLE
```

Comment:
nit: I think you'll disagree :P but I find these blocks easier to read when written with a loop:
```kotlin
val visibilitiesToCheck = arrayOf(View.VISIBLE, View.INVISIBLE, View.GONE, View.VISIBLE)
visibilitiesToCheck.forEach { visibilityToCheck ->
    killswitchLayout.visibility = visibilityToCheck
    assertEquals(View.GONE, killswitchLayout.visibility)
}
```

To me, I can quickly skim to see it says, "It asserts View.GONE for all these visibilities" rather than having to read every line because the actual code may not follow my assumptions or may be written incorrectly.

Reply:
That makes sense to me here, actually.  I'll make the change.